forc-plugins/forc-doc/src/doc.rs

```diff
@@ -49,17 +49,27 @@ impl Document {
         }
     }
     fn preview_opt(&self) -> Option<String> {
-        match &self.raw_attributes {
-            Some(description) => {
-                if description.len() > 200 {
-                    let (first, _) = description.split_at(75);
-                    Some(format!("{}...", first.trim_end()))
-                } else {
-                    Some(description.to_string())
+        self.raw_attributes.as_ref().map(|description| {
+            let preview = split_at_markdown_header(description);
+            if preview.len() > 100 && preview.contains("</p>") {
```

Comment:
We should make these magic values like `100` and `5` consts, e.g. `const PREVIEW_LEN_LIMIT: usize = 100;`, especially for those that are used multiple times.

Also, keep in mind that checking `len` only checks the *byte* length. The byte length is often not the best way to check visible length, as in utf8 some characters are represented with more bytes than others. Often it's a bit more reliable to check against `.chars().count()`, e.g.
```suggestion
            if preview.chars().count() > MAX_PREVIEW_CHARS && preview.contains("</p>") {
```